mne/export/_edf.py

```diff
@@ -189,31 +189,47 @@ def _export_raw(fname, raw, physical_range, add_ch_type):
         # set patient info
         subj_info = raw.info.get("subject_info")
         if subj_info is not None:
-            birthday = subj_info.get("birthday")
-
             # get the full name of subject if available
-            first_name = subj_info.get("first_name")
-            last_name = subj_info.get("last_name")
-            first_name = first_name or ""
-            last_name = last_name or ""
-            joiner = ""
-            if len(first_name) and len(last_name):
-                joiner = " "
-            name = joiner.join([first_name, last_name])
+            first_name = subj_info.get("first_name", "")
+            middle_name = subj_info.get("middle_name", "")
+            last_name = subj_info.get("last_name", "")
+            name = ""
+            for value in [first_name, middle_name, last_name]:
+                if value:
+                    if name:
+                        name += " "
+                    name += value
```

Comment:
I think you could do something like:

```suggestion
            name = " ".join([first_name, middle_name, last_name])
```

Reply:
Not if `first_name` or (`first_name`, `middle_name`) are empty, otherwise you simply add spaces at the beginning of `name` for no reason.
For instance:
```python
>>> first_name = ""
>>> middle_name = ""
>>> last_name = "Doe"
>>> print(" ".join([first_name, middle_name, last_name]))
'  Doe'
```
I did not find a smarter/shorter solution

Reply:
Closed by 7179a30c0052fc90b47ea5c4540cff301ac9e6d9